src/gui/mzroll/backgroundopsthread.cpp

```diff
@@ -267,8 +279,14 @@ void BackgroundOpsThread::computePeaks()
         peakDetector->mavenParameters()->pullIsotopesFlag = hadPullIsotopes;
         peakDetector->mavenParameters()->searchAdducts = hadSearchAdducts;
     }
+    
+    if (mainwindow->mavenParameters->classifyUsingPeakMl) 
+        classifyGroups(mavenParameters->allgroups);
+
     emitGroups();
 
+    Q_EMIT(updateProgressBar("Done", 1, 1));
+
```

Comment:
This should not be here. The emit will be followed by another emit in the next line, effectively making it useless.